modules-js/react-fleet/src/form-elements/inputs/MemorableDateInput.test.ts

```diff
@@ -0,0 +1,188 @@
+    expect(
+      isDateValid(
+        new Date(middleDate),
+        new Date(pastDate),
+        today,
+        true,
+        undefined
+      )
+    ).toEqual('');
+  });
+
+  test(`onlyAllowPast && earliest: ${middleDate} | user entered: ${pastDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(pastDate),
+        new Date(middleDate),
+        today,
+        true,
+        undefined
+      )
+    ).not.toEqual('');
+  });
+
+  test(`onlyAllowFuture && latest: ${futureDate} | user entered: ${upcomingDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(upcomingDate),
+        today,
+        new Date(futureDate),
+        undefined,
+        true
+      )
+    ).toEqual('');
+  });
+
+  test(`onlyAllowFuture && latest: ${upcomingDate} | user entered: ${futureDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(futureDate),
+        today,
+        new Date(upcomingDate),
+        undefined,
+        true
+      )
+    ).not.toEqual('');
+  });
+
+  test(`earliest: ${pastDate} | user entered: ${upcomingDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(upcomingDate),
+        new Date(pastDate),
+        null,
+        undefined,
+        undefined
+      )
+    ).toEqual('');
+  });
+
+  test(`earliest: ${futureDate} | user entered: ${upcomingDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(upcomingDate),
+        new Date(futureDate),
+        null,
+        undefined,
+        undefined
+      )
+    ).not.toEqual('');
+  });
+
+  test(`latest: ${futureDate} | user entered: ${upcomingDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(upcomingDate),
+        null,
+        new Date(futureDate),
+        undefined,
+        undefined
+      )
+    ).toEqual('');
+  });
+
+  test(`latest: ${upcomingDate} | user entered: ${futureDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(futureDate),
+        null,
+        new Date(upcomingDate),
+        undefined,
+        undefined
+      )
+    ).not.toEqual('');
+  });
+
+  test(`between ${pastDate} and ${upcomingDate} | user entered: ${middleDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(middleDate),
+        new Date(pastDate),
+        new Date(upcomingDate),
+        undefined,
+        undefined
+      )
+    ).toEqual('');
+  });
+
+  test(`between ${pastDate} and ${middleDate} | user entered: ${upcomingDate}`, () => {
+    expect(
+      isDateValid(
+        new Date(upcomingDate),
+        new Date(pastDate),
+        new Date(middleDate),
+        undefined,
+        undefined
+      )
+    ).not.toEqual('');
+  });
+
+  test(`no limits specified | user entered: ${upcomingDate} then ${pastDate}`, () => {
+    expect(
+      isDateValid(new Date(upcomingDate), null, null, undefined, undefined)
+    ).toEqual('');
+
+    expect(
+      isDateValid(new Date(pastDate), null, null, undefined, undefined)
+    ).toEqual('');
+  });
```

Comment:
I notice you're missing coverage for `onlyAllowFuture` or `onlyAllowPast` without any other dates specified.

(TBH you could consider dropping the `onlyAllowFuture`/`onlyAllowPast` arguments for this function in favor of just the `earliest`/`latest` `Date`s. I know that `onlyAllowFuture`/`onlyAllowPast` might be good conveniences for the component’s Props, but here I think their only value is tweaking the error message. It would simplify things.)